### PR TITLE
MO: Viewed products module now fully working

### PR DESCRIPTION
**Pull request by kpodemski**
Well, module at current state doesn't work at all, this PR makes it working, you can display viewed products anywhere you want as this is a widget

### CODE REVIEW

File: ps_viewedproduct.php [hunk 1]
```diff
@@ -168,8 +168,8 @@ public function getCacheId($name = null)
 
     public function renderWidget($hookName = null, array $configuration = [])
     {
-        if ('displayProductButtons' === $hookName) {
```
Comment (Quetzacoalt91) [inline ps_viewedproduct.php]: (I was writing another review, then I found the actual issue)

On PrestaShop 1.7.1.0, we declared an alias for the hook 'displayProductButtons'.
The only you have to replace in this module is this line with the following one:

```php
if (in_array($hookName, array('displayProductButtons', 'displayProductAdditionalInfo')))
```

**Comment by kpodemski** [comment]
@Quetzacoalt91 yeah but if you want to display module in leftColumn you can't because of this -_-

**Comment by Quetzacoalt91** [comment]
Are you sure? :)
I just added the registerHook('displayLeftColumn'), and tried to look at the category page just with my change:
capture du 2017-09-19 15-07-15

**Comment by kpodemski** [comment]
@Quetzacoalt91 with dev_mode set to true it will return a notice as you want to have productId when it's not needed :) this is why there's `else` condition in my code and i'm not looking for a vars from $configuration which is not really a good way IMO

**Comment by Quetzacoalt91** [comment]
Woops, One of my comments got lost. Let me re-add it.

File: ps_viewedproduct.php [hunk 2]
```diff
@@ -192,7 +192,11 @@ public function renderWidget($hookName = null, array $configuration = [])
 
     public function getWidgetVariables($hookName = null, array $configuration = [])
     {
-        $products = $this->getViewedProducts($configuration['product']['id_product']);
+        if ('product' === $this->context->controller->php_self) {
+            $products = $this->getViewedProducts($this->context->controller->getProduct()->id);
+        } else {
+            $products = $this->getViewedProducts(false);
+        }
```
Comment (Quetzacoalt91) [inline ps_viewedproduct.php]: For this check, I was also using the configuration variable.

$id_product = isset($configuration['product']['id_product']) ? $configuration['product']['id_product'] : false;$products = $this->getViewedProducts($id_product);

**Comment by Quetzacoalt91** [comment]
I do not understand why relying on the current function param is a bad idea for you. IMO it is even worse to look outside.

**Comment by kpodemski** [comment]
Yeah well we can argue about that, the thing is that we worked this out and module needs to be changed because right now it doesn't work :D

**Comment by idem84** [comment]
Hello, this code will fix this module, with that fix module work great on 1.7.2.4:

    public function renderWidget($hookName = null, array $configuration = [])
    {
        if (in_array($hookName, array('displayProductButtons', 'displayProductAdditionalInfo', 'displayFooterProduct'))) {
            if ('product' === $this->context->controller->php_self) {
                $id_product = $this->context->controller->getProduct()->id;
                $vieved = explode(',', $this->context->cookie->viewed);
                if (!in_array($id_product, $vieved)) {
                    $this->addViewedProduct($id_product);
                    return;
                }
            }
        }

        if (!isset($this->context->cookie->viewed) || empty($this->context->cookie->viewed)) {
            return;
        }

        $variables = $this->getWidgetVariables($hookName, $configuration);

        if (empty($variables)) {
            return false;
        }

        $this->smarty->assign($variables);

        return $this->fetch($this->templateFile);
    }

**Comment by Quetzacoalt91** [comment]
Hi @kpodemski,

As we merged the PR #4, which also fix that issue, I'm closing this one.

FYI the version v1.1.0 of the module will be deployed tomorrow.

Regards

**Comment by Juiliard** [comment]
> Hello, this code will fix this module, with that fix module work great on 1.7.2.4:
> 
> ```
> public function renderWidget($hookName = null, array $configuration = [])
> {
>     if (in_array($hookName, array('displayProductButtons', 'displayProductAdditionalInfo', 'displayFooterProduct'))) {
>         if ('product' === $this->context->controller->php_self) {
>             $id_product = $this->context->controller->getProduct()->id;
>             $vieved = explode(',', $this->context->cookie->viewed);
>             if (!in_array($id_product, $vieved)) {
>                 $this->addViewedProduct($id_product);
>                 return;
>             }
>         }
>     }
> 
>     if (!isset($this->context->cookie->viewed) || empty($this->context->cookie->viewed)) {
>         return;
>     }
> 
>     $variables = $this->getWidgetVariables($hookName, $configuration);
> 
>     if (empty($variables)) {
>         return false;
>     }
> 
>     $this->smarty->assign($variables);
> 
>     return $this->fetch($this->templateFile);
> }
> ```

works on PS 1.7.8.7 (after upgrade from 1.7.6.5)

**Pull request closed without merging**